bundles/org.openhab.binding.sony/src/main/java/org/openhab/binding/sony/internal/AbstractThingHandler.java

```diff
@@ -0,0 +1,281 @@
+     */
+    protected abstract void refreshState();
+
+    /**
+     * Returns the configuration cast to the specific type
+     * 
+     * @return a non-null configuration
+     */
+    protected C getSonyConfig() {
+        return getConfigAs(configType);
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN, ThingStatusDetail.NONE, "Initializing ...");
+        SonyUtil.cancel(retryConnection.getAndSet(this.scheduler.submit(() -> {
+            doConnect();
+        })));
+    }
+
+    /**
+     * Helper method to start a connection attempt
+     */
+    private void doConnect() {
+        updateStatus(ThingStatus.UNKNOWN, ThingStatusDetail.NONE, "Initializing ...");
+        connect();
+    }
+
+    @Override
+    protected void updateStatus(final ThingStatus status, final ThingStatusDetail statusDetail,
+            final @Nullable String description) {
+        super.updateStatus(status, statusDetail, description);
+
+        if (status == ThingStatus.ONLINE) {
+            schedulePolling();
+            scheduleCheckStatus();
+        } else if (status == ThingStatus.UNKNOWN) {
+            // probably in the process of reconnecting - ignore
+            logger.trace("Ignoring thing status of UNKNOWN");
+        } else {
+            SonyUtil.cancel(refreshState.getAndSet(null));
+            SonyUtil.cancel(checkStatus.getAndSet(null));
+
+            // don't bother reconnecting - won't fix a configuration error
+            if (statusDetail != ThingStatusDetail.CONFIGURATION_ERROR) {
+                scheduleReconnect();
+            }
+        }
+    }
+
+    /**
+     * Starts the polling process. The polling process will refresh the state of the sony device if the refresh time (in
+     * seconds) is greater than 0. This process will continue until cancelled.
+     */
+    private void schedulePolling() {
+        final C config = getSonyConfig();
+        final Integer refresh = config.getRefresh();
+
+        if (refresh != null && refresh > 0) {
+            logger.info("Starting state polling every {} seconds", refresh);
+            SonyUtil.cancel(refreshState.getAndSet(this.scheduler.scheduleWithFixedDelay(() -> {
+                // throw exceptions to prevent future tasks under these circumstances
+                if (isRemoved()) {
+                    throw new IllegalStateException("Thing has been removed - ending state polling");
+                }
+                if (SonyUtil.isInterrupted()) {
+                    throw new IllegalStateException("State polling has been cancelled");
+                }
+
+                // catch the various runtime exceptions that may occur here (the biggest being ProcessingException)
+                // and handle it.
+                try {
+                    if (thing.getStatus() == ThingStatus.ONLINE) {
+                        refreshState();
+                    }
+                } catch (final Exception ex) {
```

Comment:
I would make sure to catch specific Exceptions. You wouldn't want to mishandle a RuntimeException like an OutOfMemoryException.